### ISSUE TITLE
[sig-cluster-lifecycle] Reboot [Disruptive] [Feature:Reboot] each node by ordering unclean reboot and ensure they function upon restartChanges

### ISSUE THREAD

**Issue by mortent** [kubernetes/kubernetes#69786]
Reboot tests have been failing since 10/12

Dashboard: https://k8s-testgrid.appspot.com/sig-release-master-blocking#gke-cos-master-reboot

Sample Failure: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gke-reboot/18153

This test failure will block 1.13 beta, so please let's fix it before then. Thanks!

/sig cluster-lifecycle
/kind failing-test
/priority important-soon

**Comment by jberkus** [comment]
Note: this failure is similar to #63612 but is not the same failure, according to @yujuhong 

This is a consistent failure rather than a flake, and all nodes are failing to reboot.

It's also been failing consistently since 10/11.

/cc @msau42

**Comment by msau42** [comment]
Copying my notes here:

Looking at one recent test run:
https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gke-reboot/18231/build-log.txt

The test failed because kubelet never became "not ready":
```
I1015 19:33:01.392] Oct 15 19:33:01.390: INFO: Waiting up to 2m0s for node gke-e2e-18231-344e3-default-pool-2b8d22c7-njwh condition Ready to be false
I1015 19:33:01.393] Oct 15 19:33:01.393: INFO: Condition Ready of node gke-e2e-18231-344e3-default-pool-2b8d22c7-njwh is true instead of false. Reason: KubeletReady, message: kubelet is posting ready status. AppArmor enabled
...
I1015 19:34:59.825] Oct 15 19:34:59.825: INFO: Condition Ready of node gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr is true instead of false. Reason: KubeletReady, message: kubelet is posting ready status. AppArmor enabled
I1015 19:34:59.826] Oct 15 19:34:59.825: INFO: Condition Ready of node gke-e2e-18231-344e3-default-pool-2b8d22c7-q5g8 is true instead of false. Reason: KubeletReady, message: kubelet is posting ready status. AppArmor enabled
I1015 19:34:59.826] Oct 15 19:34:59.825: INFO: Condition Ready of node gke-e2e-18231-344e3-default-pool-2b8d22c7-njwh is true instead of false. Reason: KubeletReady, message: kubelet is posting ready status. AppArmor enabled
I1015 19:35:01.825] Oct 15 19:35:01.825: INFO: Node gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr didn't reach desired Ready condition status (false) within 2m0s
I1015 19:35:01.826] Oct 15 19:35:01.825: INFO: Node gke-e2e-18231-344e3-default-pool-2b8d22c7-q5g8 didn't reach desired Ready condition status (false) within 2m0s
I1015 19:35:01.830] Oct 15 19:35:01.826: INFO: Node gke-e2e-18231-344e3-default-pool-2b8d22c7-njwh didn't reach desired Ready condition status (false) within 2m0s
I1015 19:35:01.830] Oct 15 19:35:01.826: INFO: Node gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr failed reboot test.
I1015 19:35:01.830] Oct 15 19:35:01.826: INFO: Node gke-e2e-18231-344e3-default-pool-2b8d22c7-njwh failed reboot test.
I1015 19:35:01.830] Oct 15 19:35:01.826: INFO: Node gke-e2e-18231-344e3-default-pool-2b8d22c7-q5g8 failed reboot test.
```

When looking at one of the kubelet logs, I do see that kubelet had problems making Node updates to the api server:
https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gke-reboot/18231/artifacts/gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr/kubelet.log
```
E1015 19:33:26.150380    1303 kubelet_node_status.go:377] Error updating node status, will retry: error getting node "gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr": Get https://35.202.63.115/api/v1/nodes/gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr?resourceVersion=0&timeout=10s: context deadline exceeded (Client.Timeout exceeded while awaiting headers)
I1015 19:33:29.552319    1303 cloud_request_manager.go:113] Node addresses from cloud provider for node "gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr" not collected
E1015 19:33:36.150912    1303 kubelet_node_status.go:377] Error updating node status, will retry: error getting node "gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr": Get https://35.202.63.115/api/v1/nodes/gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr?timeout=10s: context deadline exceeded (Client.Timeout exceeded while awaiting headers)
I1015 19:33:41.553161    1303 cloud_request_manager.go:113] Node addresses from cloud provider for node "gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr" not collected
E1015 19:33:46.152035    1303 kubelet_node_status.go:377] Error updating node status, will retry: error getting node "gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr": Get https://35.202.63.115/api/v1/nodes/gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr?timeout=10s: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
I1015 19:33:53.554253    1303 cloud_request_manager.go:113] Node addresses from cloud provider for node "gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr" not collected
E1015 19:33:56.152555    1303 kubelet_node_status.go:377] Error updating node status, will retry: error getting node "gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr": Get https://35.202.63.115/api/v1/nodes/gke-e2e-18231-344e3-default-pool-2b8d22c7-01rr?timeout=10s: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

Errors continue until kubelet is restarted 8 minutes later (the test times out at 2 minutes):
```
Stopping Kubernetes kubelet...
Stopped Kubernetes kubelet.
Started Kubernetes kubelet.
Flag --allow-privileged has been deprecated, will be removed in a future version
Flag --max-pods has been deprecated, This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.
I1015 19:41:30.505574    1169 flags.go:33] FLAG: --address="0.0.0.0"
```

**Comment by msau42** [comment]
So it seems like even though kubelet is unable to update its Node status, the Node is still showing up as Ready

**Comment by jberkus** [comment]
Adding sig-node to this discussion:

/sig node

**Comment by msau42** [comment]
I'm trying to repro on a dev 1.11/1.12 cluster so we can get master logs for the node controller

**Comment by msau42** [comment]
@kubernetes/sig-node-bugs

**Comment by AishSundar** [comment]
/cc @yujuhong 

adding @yujuhong from sig-node

**Comment by yujuhong** [comment]
The node controller doesn't seem to be working at all as it wasn't marking the nodes not ready.

@msau42 suggested that the api group may not have been whitelisted in GKE, so I checked and that seemed to be true. We can whitelist it, but for an alpha feature, it should not be properly gated and not affect production.

We can revert PR #69241 to unblock release, and wait for @wangzhen127 to re-submit a patched PR.

/cc @wojtek-t

**Comment by wangzhen127** [comment]
Thanks for debugging this! Just got time to take a quick look. Does it mean [creating the lease lister](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L390) should also be put behind the feature gate?

**Comment by wojtek-t** [comment]
I think reverting is a good option to unblock stuff.

@wangzhen127 - creating listed may not be necessary, but setting HasSynced is (so if we have to put it after feature gate, we should put both).

**Comment by wangzhen127** [comment]
Is there any guideline on how to revert my original PR #69241? I haven't done it before. Just revert in my fork and submit a new PR?

I also just put up a fix PR #69933. I am fine with either submitting the fix PR or revert my original PR first.

**Comment by wangzhen127** [comment]
/reopen
/assign @wangzhen127 
I will manually close this after verifying the bots.

**Comment by AishSundar** [comment]
This is passing in the latest test run. Thanks @wangzhen127 for the quick turnaround

**Comment by wangzhen127** [comment]
Nice! Closing this now.
/close
